telemetry/tracks/tracks-utils.php

```diff
@@ -0,0 +1,114 @@
+<?php
+/**
+ * Telemetry: Tracks Utils
+ *
+ * @package Automattic\VIP\Telemetry\Tracks
+ */
+
+declare(strict_types=1);
+
+namespace Automattic\VIP\Telemetry\Tracks;
+
+use Automattic\VIP\Support_User\User as Support_User;
+
+/**
+ * Returns the base properties for a track event.
+ *
+ * @return array<string, mixed> The base properties.
+ */
+function get_base_properties_of_track_event(): array {
+	$props = [
+		'hosting_provider' => get_hosting_provider(),
+		'is_vip_user'      => Support_User::user_has_vip_support_role( get_current_user_id() ),
+		'is_multisite'     => is_multisite(),
+		'wp_version'       => get_bloginfo( 'version' ),
+	];
+
+	// Set VIP environment if it exists.
+	if ( defined( 'VIP_GO_APP_ENVIRONMENT' ) ) {
+		$app_environment = constant( 'VIP_GO_APP_ENVIRONMENT' );
+		if ( is_string( $app_environment ) && '' !== $app_environment ) {
+			$props['vipgo_env'] = $app_environment;
+		}
+	}
+
+	// Set VIP organization if it exists.
+	if ( defined( 'VIP_ORG_ID' ) ) {
+		$org_id = constant( 'VIP_ORG_ID' );
+		if ( is_integer( $org_id ) && $org_id > 0 ) {
+			$props['vipgo_org'] = $org_id;
+		}
+	}
+
+	return $props;
+}
+
+/**
+ * Check if the site is hosted on VIP.
+ */
+function is_wpvip_site(): bool {
+	return defined( 'WPCOM_IS_VIP_ENV' ) && constant( 'WPCOM_IS_VIP_ENV' ) === true
+		&& defined( 'WPCOM_SANDBOXED' ) && constant( 'WPCOM_SANDBOXED' ) === false;
+}
+
+/**
+ * Get the hosting provider.
+ */
+function get_hosting_provider(): string {
+	if ( is_wpvip_site() ) {
+		return 'wpvip';
```

Comment:
`is_wpvip_site()` returns `false` for a sandboxed site. Is that correct behavior for this to not say `wpvip` is the host if it's sandboxed?

Reply:
Nice catch. Updated the implementation of `get_hosting_provider` (https://github.com/Automattic/vip-go-mu-plugins/pull/5966/commits/fe56a14ab068e270a49aba01a7987ba2c41d6b20) and now it returns `wpvip_sandbox` for sandboxed sites.